File: server/internal/world/star_system.go
package world

import (
	"srv/internal/utils/phys"
)

type StarParams struct {
	Temperature phys.Temperature
	Luminosity  phys.Luminosity
	Mass        phys.Mass
	Radius      phys.Distance
	Age         phys.Age
}

type Star struct {
	ID     CelestialID
	Params StarParams
}

type StarSystemOverview struct {
	ID     StarSystemID
	Coords GalacticCoords

	IsExplored bool
	Stars      []Star
	NPlanets   int
	NMoons     int
	NAsteroids int
	PopInfo    PopulationOverview
}

type StarSystemContent struct {
	ID StarSystemID

	Explored ExplorationData

	Orbits map[CelestialID]OrbitData
	Stars  []Star
}

File: ui/src/domain/StarSystem.ts
import { type GalacticCoords } from './GalacticCoords';
import { type Orbit } from './Orbit';
import { type Star } from './Star';
import { type ExplorationData } from './misc';

export type StarSystemOverview = {
    id: string;
    coords: GalacticCoords;
    stars: Star[];
    nPlanets: number;
    nAsteroids: number;
    isExplored: boolean;
};

export type StarSystemContent = {
    id: string;
    orbits: Record<string, Orbit>;
    stars: Star[];
    explored: ExplorationData | null;
};
